packages/backend/src/api/jobScraping/scraping.ts

```diff
@@ -0,0 +1,349 @@
+import * as cheerio from 'cheerio';
+import puppeteer from 'puppeteer';
+
+function filterLanguagesAndFrameworks(paragraph: string) {
+  const languages = [
+    'JavaScript',
+    'TypeScript',
+    'Objective-C',
+    'C++',
+    'C#',
+    'Python',
+    'Java',
+    'Ruby',
+    'PHP',
+    'Swift',
+    'Go',
+    'Rust',
+    'Kotlin',
+    'Dart',
+    'R',
+    'C',
+    'Perl',
+    'Scala',
+    'Haskell',
+  ];
+
+  const frameworks = [
+    'React',
+    'Vue.js',
+    'Angular',
+    'Svelte',
+    'Next.js',
+    'Express',
+    'Django',
+    'Flask',
+    'Spring Boot',
+    'Ruby on Rails',
+    'Laravel',
+    'Symfony',
+    'ASP.NET',
+    'Nuxt.js',
+    'Nest.js',
+    'Flutter',
+    'React Native',
+    'Electron',
+    'Meteor',
+    'Backbone.js',
+    'Ember.js',
+    'Fast API',
+    'Bootstrap',
+    '.NET Core',
+  ];
+
+  // Ensure special characters are escaped (C++, C#)
+  const escapedLanguages = languages.map((lang) => lang.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'));
+  const escapedFrameworks = frameworks.map((framework) => framework.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'));
+
+  // Join into a regex that matches any of the languages or frameworks
+  const languagesRegex = new RegExp(`(?<![\\w])(${escapedLanguages.join('|')})(?![\\w])`, 'gi');
+  const frameworksRegex = new RegExp(`(?<![\\w])(${escapedFrameworks.join('|')})(?![\\w])`, 'gi');
+
+  const foundLanguages = paragraph.match(languagesRegex) || [];
+  const foundFrameworks = paragraph.match(frameworksRegex) || [];
+
+  return {
+    languages: [...new Set(foundLanguages)], // Removing duplicates
+    frameworks: [...new Set(foundFrameworks)], // Removing duplicates
+  };
+}
+
+async function shortenLink(link: any) {
+  const shortenedLink = await tinyURLShorten(link);
+  return shortenedLink;
+}
+
+async function tinyURLShorten(link: any) {
+  const response = await fetch(`https://tinyurl.com/api-create.php?url=${encodeURIComponent(link)}`);
```

Comment:
tiny url's free tier limits us to generate 100 urls max, why are we shortening the urls using tinyUrl?

Reply:
For now we are using tinyurl but we will not later on, I already talk with hasan about this. This just to remember we need to short the urls